fix: Profile - username and Position not updated directly in the profile's header after add/edit - EXO-62664 - Meeds-io/meeds#756

GouadriaHanen opened the pull request:
prior to this change, after saving/editing the username or user position are not updated directly in the profile's header 
since the data from the event 'userModified' in this case is null
after this change, the modification is well displaying by refreshing the user properties

aycherif committed:
TASK-62301: Create FB maintenance and update projects versions/dependencies

IlhemEssaadi committed:
fix: fix space and user popover position - EXO-62433 - Meeds-io/meeds#706 (#2330)

Prior to this fix, the space and user popover hide the user or space name and when the popover is displayed and we scroll down or up. The popover remains displayed even if the mouse is no longer on the name of the space because there is waiting time before the popover closes. After this fix, space and user popover is positioned under the user or space name and when we scroll up or down, the popover disappears and closes immediately.

ahamdi committed:
fix: error 500 when login with Openid - EXO-62366 - Meeds-io/meeds#750 (#2348)

Prior to this fix, with Openid authentication configured, when logging
on Meeds server an error page (HTTP 500) is displayed.
The SocialUserProfileEventListenerImpl was retrieving services from the
Root container and a NullPointerException was thrown.
The fix will encapsulate the code executed in the
SocialUserProfileEventListenerImpl inside a requestlifecycle with the
correct Portal container.

GouadriaHanen committed:
fix: Profile - Position not updated directly in the profile's header after add/edit - EXO-62664

rdenarie commented:
Should we passe the object user in the detail of the event ?

ahamdi committed:
fix: avoid displaying and indexing user technical properties - EXO-62724 - meeds-io/meeds#762 (#2360)

When login using openID provider, the related user informations are
saved in the user profile an then indexed in ElasticSearch which causes
an indexing error. Those properties are also displayed in the user
profile while they are technical information that should not be viewed.
The fix sets user properties to inactive (to not display them in the
user profile) and do not index them until they are activated by the
administrator from the Profile properties settings.

mkrout committed:
fix: Unable to add new IMs when updating the field - EXO-62687 - Meeds-io/meeds#761 (#2359)

Prior to this change when the user tries to edit ims or phones property
in his profile, only already filled options are displayed
After this modification all options are displayed in the edit drawer of
profile for IMS and Phones properties

ahamdi commented on webapp/portlet/src/main/webapp/vue-apps/profile-header/components/ProfileHeader.vue:
```diff
@@ -146,8 +146,10 @@ export default {
       if (event && event.detail && event.detail !== this.user) {
         this.user = Object.assign({}, this.user, event.detail);
-        this.$nextTick().then(() => this.$root.$emit('application-loaded'));
+      } else  {
```
This is a forcing to reload that is not needed. Could you check why the userModified is null ?

GouadriaHanen replied:
the event is dispatched by ProfileContactInformationDrawer which has only the user's properties to modify. 
https://github.com/Meeds-io/social/blob/7f85b69e9c7edcbbfed6485087f05eee94c865f7/webapp/portlet/src/main/webapp/vue-apps/profile-contact-information/components/ProfileContactInformationDrawer.vue#L205

GouadriaHanen replied:
FYI @rdenarie

ahamdi replied:
@GouadriaHanen the check inside the event listener should be updated without the need to call refresh explicitly which will reload data from server.
you can convert the array of properties in https://github.com/Meeds-io/social/blob/7f85b69e9c7edcbbfed6485087f05eee94c865f7/webapp/portlet/src/main/webapp/vue-apps/profile-contact-information/components/ProfileContactInformationDrawer.vue#L205 to a user object  then send it in the event data . this way you won't need to change in [ProfileHeader.vue](https://github.com/Meeds-io/social/pull/2353/files#diff-8a7df92ac03c601f3d8d0748d9960ec49280076846328494cb5c16bfe7b57256)

GouadriaHanen replied:
done!

IlhemEssaadi committed:
fix: fix space and user popover position - EXO-62433 - Meeds-io/meeds#706

Prior to this change, the space popover would display, but hide the space name and when I place the cursor over the space/user popover and scroll down the page, the popover remains displayed. After this change, the popover is displayed under the space name in case the top space is not enough to display it and we force the user popover to close when scrolling.

IlhemEssaadi committed:
fix :fix save button not disabled in profile information drawer when there is no change - EXO-62627 - Meeds-io/meeds#763

Prior to this change, when we access to edit the profile contact informations without changing anything, the save button is proposed not disabled and not clickable, so the edit drawer cannot be closed. After this fix, the save button is disabled when we open the drawer and no change is detected, and it is enabled when we detect a change in any of the profile information.

GouadriaHanen committed:
broadcast event with data user

ahamdi committed:
fix: ignore openId properties while synchronize user profile - EXO-62724 - meeds-io/meeds#762 (#2366)

Some technical properties saved in Portal user profile are used for
technical purposes and not intended to be shared with the social user
profile. Those properties contain DOT '.' character which may cause
problems when indexing those properties.
The fix adds the possibility to define properties to ignore when
synchronizing both profiles and replaces dot '.' characters with
underscore '_' for indexed filed names .

mkrout committed:
fix: Unable to add new IMs when updating the field - EXO-62687 - Meds-io/meeds#761 (#2370)

Prior to this change when the user tries to edit ims or phones property
in his profile, only already filled options are displayed
After this modification all options are displayed in the edit drawer of
profile for IMS and Phones properties

GouadriaHanen committed:
refactor getting user properties

ahamdi committed:
Merge branch 'feature/maintenance' into EXO-62664

ahamdi committed:
Update ProfileIndexingServiceConnector.java
